### DISCUSSION TITLE
Can access t in script but not script setup?

### DISCUSSION THREAD

**Discussion by HittiGubbi** [intlify/vue-i18n#996] [Q&A] [answered]
I have a basic Vue single file component with the following kind of a script element:

    <script>
        export default {
            computed: {
                someProperty() {
                    return this.$t("some.property");
                }
            }
        }
    </script>

This works without problems: the value of `someProperty` correctly changes based on the current locale.

The problem is that I would prefer to use script setup instead, as that is what I am using everywhere else, but I have been unable to find information how to achieve that. E.g. the following two attempts result in error about t being undefined:

    <script setup>
        import { computed } from 'vue';
        const someProperty = computed(() => this.$t.("some.property"));
    </script>

    <script setup>
        import { computed } from 'vue';
        const someProperty = computed(() => t.("some.property"));
    </script>

So how can I access t inside a script setup element? I naturally want to do this in same manner as in the normal script, that is, without explicit import of t in different single file components.

**Comment by kazupon** [comment, marked as answer]
>So how can I access t inside a script setup element? I naturally want to do this in same manner as in the normal script, that is, without explicit import of t in different single file components.

You need to use the `useI18n` in script setup.
```vue
<script setup>
    import { computed } from 'vue';
    import { useI18n } from 'vue-i18n';
    const { t } = useI18n();
    const someProperty = computed(() => t.("some.property"));
</script>
```
See the details composition API section:
https://vue-i18n.intlify.dev/guide/advanced/composition.html

**Reply by HittiGubbi** [reply]
Thanks, this works. So it seems that the explicit import cannot be avoided in script setup elements.

**Reply by CaiWenlie** [reply]
but how can i use i18n in js file?

**Reply by MarcMoerkeberg** [reply]
For some reason, I thought I had to use the custom instance of `i18n` I mounted on app in `main.ts`. [The documentation however also specifies](https://vue-i18n.intlify.dev/guide/advanced/composition.html#message-translation) it should be done with `useI18n()` from the plugin library, I guess I was just stuck in my old ways of using `i18n` previously with _Vue_ and _TypeScript_.
Thank you very much for the clear and concise answer @kazupon 😄

**Reply by realtebo** [reply]
"Not available in legacy mode"

**Reply by Carlozc90** [reply]
if you have the error "Not available in legacy mode"  

const i18n = createI18n({
  legacy: false,
  ...
